parsons/dbt/dbt.py

```diff
@@ -0,0 +1,268 @@
+import logging
+import shutil
+import pathlib
+import os
+import subprocess
+from parsons.notifications.slack import Slack
+import json
+from typing import Optional, Literal, List
+from parsons.utilities import check_env
+import time
+
+logger = logging.getLogger(__name__)
+
+
+class dbtLogger:
```

Comment:
I'm more concerned with abstracting from "Redshift" to "works with all databases" but we could also abstract from "Slack" to "works with any notification system". 

To be clear, I'm not asking you to build that capability, just thinking on how we could structure this so it's easy for someone else to come build that capability. In particular, I'm thinking about [dependency injection/inversion of control](https://stackoverflow.com/questions/3058/what-is-inversion-of-control). I don't have much CS background - do you? My understanding is that in cases like these we'd want to create a logger that accepts a notification/messaging object and calls `message(` on it and a runner that accepts a database connector and calls `query(` on it but doesn't care about the implementation.

Reply:
Reading through this whole thing, I realize now that you're not really using any of the DB connectors directly so you're not calling `query(`. Instead, you're using the `dbt_redshift` dependency and sending connection info and commands directly to it. 

So the challenge becomes whether we can abstract over the `dbt_redshift` interface vs the `dbt_bigquery` interface.

Reply:
Ok, this is resolved with 088f974515e81de65d6173a78febd31e8a7cbba7, which allows for this all to work with any database engine that works with dbt. This is actually handled at the level of package dependencies rather than in the code itself. So now the dbt extra can be installed with any database-specific version of dbt. e.g. `pip install parsons[dbt-redshift]`, `pip install parsons[dbt-bigquery]` etc